### PR TITLE
Errors 2.3 - Mesh Definition

### PR DESCRIPTION
**Pull request by fsimonis**
**List the core changes of this PR**

* Reword mesh-related error messages.
* Refactor common error messages into dedicated functions

**Additional Information**

Related to #698

### CODE REVIEW

File: src/precice/impl/SolverInterfaceImpl.cpp [hunk 1]
```diff
@@ -859,10 +867,11 @@ void SolverInterfaceImpl::setMeshQuadWithEdges(
...
   if (context.meshRequirement == mapping::Mapping::MeshRequirement::FULL) {
     mesh::PtrMesh &mesh = context.mesh;
-    PRECICE_CHECK(mesh->isValidVertexID(firstVertexID), "Given VertexID is invalid!");
-    PRECICE_CHECK(mesh->isValidVertexID(secondVertexID), "Given VertexID is invalid!");
-    PRECICE_CHECK(mesh->isValidVertexID(thirdVertexID), "Given VertexID is invalid!");
-    PRECICE_CHECK(mesh->isValidVertexID(fourthVertexID), "Given VertexID is invalid!");
+    using impl::errorInvalidVertexID;
+    PRECICE_CHECK(mesh->isValidVertexID(firstVertexID), errorInvalidVertexID(firstVertexID));
+    PRECICE_CHECK(mesh->isValidVertexID(secondVertexID), errorInvalidVertexID(secondVertexID));
+    PRECICE_CHECK(mesh->isValidVertexID(thirdVertexID), errorInvalidVertexID(thridVertexID));
```
Comment (uekerman) [inline src/precice/impl/SolverInterfaceImpl.cpp:873]:
```suggestion
    PRECICE_CHECK(mesh->isValidVertexID(thirdVertexID), errorInvalidVertexID(thirdVertexID));
```

File: src/partition/ProvidedPartition.cpp [hunk 2]
```diff
@@ -104,7 +104,8 @@ void ProvidedPartition::communicate()
       Event e("partition.sendGlobalMesh." + _mesh->getName(), precice::syncMode);
 
       if (not utils::MasterSlave::isSlave()) {
-        PRECICE_CHECK(globalMesh.vertices().size() > 0, "The provided mesh " << globalMesh.getName() << " is invalid (possibly empty).");
+        PRECICE_CHECK(globalMesh.vertices().size() > 0,
+                      "The provided mesh \"" << globalMesh.getName() << "\" was not defined. Please set the mesh using setMeshXXX() prior to calling initialize().");
```
Comment (uekerman) [inline src/partition/ProvidedPartition.cpp:108]:
```suggestion
                      "The provided mesh \"" << globalMesh.getName() << "\" is empty. Please set the mesh using setMeshXXX() prior to calling initialize().");
```
"defined" is a bit ambiguous. We could say that one defines a mesh already in the config.

File: src/partition/ReceivedPartition.cpp [hunk 3]
```diff
@@ -426,11 +420,10 @@ void ReceivedPartition::compareBoundingBoxes()
 
     // send connectionMap to other master
     m2n().getMasterCommunication()->send(connectedRanksList, 0);
-    if (not connectionMap.empty()) {
-      com::CommunicateBoundingBox(m2n().getMasterCommunication()).sendConnectionMap(connectionMap, 0);
-    } else {
-      PRECICE_ERROR("This participant seems to have no mesh partitions for mesh " << _mesh->getName() << " at the coupling interface.");
-    }
+    PRECICE_CHECK(not connectionMap.empty(), "The mesh \"" << _mesh->getName() << "\" of this participant has no partitions at the coupling interface. "
```
Comment (uekerman) [inline src/partition/ReceivedPartition.cpp:423]:
```suggestion
    PRECICE_CHECK(not connectionMap.empty(), "The mesh \"" << _mesh->getName() << "\" of this participant seems to have no partitions at the coupling interface. "
```
It depends who defines where the coupling interface is. This participant or the other one? I prefer the weaker statement here.

File: src/partition/ReceivedPartition.cpp [hunk 4]
```diff
@@ -426,11 +420,10 @@ void ReceivedPartition::compareBoundingBoxes()
...
+                                                                                  "Check that both mapped meshes are describing the same geometry. "
+                                                                                  "If you deal with very coarse meshes, consider increasing the safety-factor in the <use-mesh /> tag.");
```
Comment (uekerman) [inline src/partition/ReceivedPartition.cpp:425]:
```suggestion
                                                                                  "If you deal with very different mesh resolutions, consider increasing the safety-factor in the <use-mesh /> tag.");
```
Just being coarse is no problem I guess?

File: src/partition/ProvidedPartition.cpp (hunk 2 above)
Comment (fsimonis) [reply src/partition/ProvidedPartition.cpp:108]: I am fine with the change.

Normally declaration means "there is a mesh" and definition means "this is how the mesh looks".
But my sight is probably tainted by the programmer inside me.

File: src/partition/ReceivedPartition.cpp (hunk 3 above)
Comment (fsimonis) [reply src/partition/ReceivedPartition.cpp:423]: This check is in the master block of the `ReceivedPartition`, so the side should be clear here.

File: src/partition/ReceivedPartition.cpp (hunk 4 above)
Comment (fsimonis) [reply src/partition/ReceivedPartition.cpp:425]: Your version is what I was looking for :wink:

File: src/partition/ReceivedPartition.cpp (hunk 3 above)
Comment (uekerman) [reply src/partition/ReceivedPartition.cpp:423]: What I meant is: who defines where the coupling interface is? This participant or the other one? The error here means in the end that both sides probably defined different coupling interfaces (concerning the geometry). Maybe we should just reformulate the first sentence completely. 
"Re-partitioning the received mesh XXX resulted in no overlaps with provided meshes."
The suggestions afterwards are good.

**Pull request merged**